nservicebus/transactional-session/index.md

```diff
@@ -51,17 +50,45 @@ Sending messages in an atomic manner is done through the `ITransactionalSession`
 
 snippet: sending-transactional-session
 
-The persistence specific database session is accessible via the `transactionalSession.SynchronizedStorageSession` property or via dependency injection. See the [persistence specific documentation](/nservicebus/transactional-session/persistences) for more details.
+The persistence-specific database session is accessible via the `transactionalSession.SynchronizedStorageSession` property or via dependency injection. See the [persistence-specific documentation](/nservicebus/transactional-session/persistences) for more details.
 
 Once all the operations that are part of the atomic request have been executed, the session should be committed:
 
 snippet: committing-transactional-session
 
-Disposing the transactional session without committing, will roll-back any changes that were made.
+Disposing the transactional session, without committing, will roll back any changes that were made.
+
+### Advanced
+
+#### Maximum commit duration
+
+The maximum commit duration defaults to `TimeSpan.FromSeconds(15)`.
+
+The maximum commit duration limits the amount of time the transaction can attempt to commit the changes atomically. The value can be configured when opening the session.
+
+snippet: configuring-timeout-transactional-session
+
+The maximum commit duration does not represent the total transaction time, but rather the actual time observed from the perspective of the control message used to settle the transaction outcome. The total transaction time observed might be longer, considering delays in the transport due to latency, delayed delivery, load on the input queue, endpoint concurrency limits, and more.
+
+When the control message is consumed, but the outbox record is not yet available in storage, the following formula is applied to delay the message (see [Phase 2](#how-it-works-phase-2)):
+
+```text
+CommitDelayIncrement = 2 * CommitDelayIncrement;
+RemainingCommitDuration = RemainingCommitDuration - 
+   (CommitDelayIncrement > RemainingCommitDuration ? RemainingCommitDuration : CommitDelayIncrement)
+```
+
+The default commit delay increment is set to `Timespan.FromSeconds(2)`and cannot be overridden.
+
+#### Metadata
+
+It is possible to add custom headers to the control message that is used to settle the transaction outcome. Custom headers can be set by adding them to the metadata when opening the session.
```

Comment:
Should we add information why would you ever want to do so? Like

```suggestion
It is possible to add custom headers to the control message that is used to settle the transaction outcome. Custom headers can be set by adding them to the metadata when opening the session. These headers can be access by a [custom behavior](https://docs.particular.net/nservicebus/pipeline/manipulate-with-behaviors#add-a-new-step) in the `TransportReceive` part of the pipeline.
```

Reply:
I was thinking about that and then I dismissed it because I felt that probably you never ever want to manipulate these control messages. Happy though to accept the suggestion